### PR TITLE
Tape and nock and updated dependencies

### PR DESCRIPTION
**Pull request by Flet**
Sorry I realize this is a big diff :(

- Refreshed dependencies.
- Removed unused dotfiles.
- Switched from jasmine tests to tape tests.
- Added nock fixtures to mock all API requests.
  - Also took care to ensure no apiKeys are exposed in fixture .json files.

- Fixed a bug! non 2XX http responses should be throwing errors when using promises. 
 - There were some tests looking for this, but they never got executed because the catch() was never called.

### CODE REVIEW

File: lib/Resource.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 var request = require('request-promise');
-var _ = require('lodash');
+var assign = require('lodash.assign');
```
Comment (Flet) [inline lib/Resource.js]: Switched to lodash.assign as its lighter than including all of lodash

File: lib/Resource.js [hunk 2]
```diff
@@ -77,18 +77,18 @@ var Resource = function (_strategy, _config) {
...
+        if (callback) {
+          callback(null, body);
+        } else {
+          return body;
+        }
+      })
+      .catch(function (err) {
+        if (callback) {
+          callback(err);
+        } else { throw err; }
```
Comment (Flet) [inline lib/Resource.js]: Here is the bug fix; the error needs to be thrown instead of returned.

**Comment by ecaron** [comment]
Think this is a 1.1.0 release, or 1.0.1?

**Comment by Flet** [comment]
It depends on if you thing throwing instead of returning errors is a bug fix or not. 

Honestly, it could even be considered a breaking change since some folks might be looking for error statusCodes in the response.

**Comment by ecaron** [comment]
@Flet - Go with marking this a 1.0.1

**Comment by Flet** [comment]
Published as 1.0.1

**Pull request merged**